internal/util/ssa/patch_test.go

```diff
@@ -123,30 +162,57 @@ func TestPatch(t *testing.T) {
 		// Verify that gvk is still set
 		g.Expect(createObject.GroupVersionKind()).To(Equal(initialObject.GroupVersionKind()))
 
-		// 2. Update the object and verify that the request was not cached as the object was changed.
+		// 2. Update the object and verify that the request was not cached with the old identifier,
+		// but is cached with a new identifier (after apply).
 		// Get the original object.
 		originalObject := initialObject.DeepCopy()
 		g.Expect(env.GetAPIReader().Get(ctx, client.ObjectKeyFromObject(originalObject), originalObject)).To(Succeed())
 		// Modify the object
 		modifiedObject := initialObject.DeepCopy()
 		modifiedObject.Spec.Deletion.NodeDrainTimeoutSeconds = ptr.To(int32(5))
-		// Compute request identifier, so we can later verify that the update call was not cached.
+		// Compute request identifier before the update, so we can later verify that the update call was not cached with this identifier.
 		modifiedUnstructured, err := prepareModified(env.Scheme(), modifiedObject)
 		g.Expect(err).ToNot(HaveOccurred())
-		requestIdentifier, err := ComputeRequestIdentifier(env.GetScheme(), originalObject, modifiedUnstructured)
+		oldRequestIdentifier, err := ComputeRequestIdentifier(env.GetScheme(), originalObject.GetResourceVersion(), modifiedUnstructured)
 		g.Expect(err).ToNot(HaveOccurred())
+		// Save a copy of modifiedUnstructured before apply to compute the new identifier later
+		modifiedUnstructuredBeforeApply := modifiedUnstructured.DeepCopy()
 		// Update the object
-		g.Expect(Patch(ctx, env.GetClient(), fieldManager, modifiedObject, WithCachingProxy{Cache: ssaCache, Original: originalObject})).To(Succeed())
+		applyCallCount = 0
+		g.Expect(Patch(ctx, countingClient, fieldManager, modifiedObject, WithCachingProxy{Cache: ssaCache, Original: originalObject})).To(Succeed())
+		g.Expect(applyCallCount).To(Equal(1), "Expected 1 API call for first update (object changed)")
 		// Verify that gvk is still set
 		g.Expect(modifiedObject.GroupVersionKind()).To(Equal(initialObject.GroupVersionKind()))
-		// Verify that request was not cached (as it changed the object)
-		g.Expect(ssaCache.Has(requestIdentifier, initialObject.GetObjectKind().GroupVersionKind().Kind)).To(BeFalse())
+		// Verify that request was not cached with the old identifier (as it changed the object)
+		g.Expect(ssaCache.Has(oldRequestIdentifier, initialObject.GetObjectKind().GroupVersionKind().Kind)).To(BeFalse())
+		// Get the actual object from server after apply to compute the new request identifier
+		objectAfterApply := initialObject.DeepCopy()
+		g.Expect(env.GetAPIReader().Get(ctx, client.ObjectKeyFromObject(objectAfterApply), objectAfterApply)).To(Succeed())
+		// Convert to unstructured WITHOUT filtering to preserve server fields (like resourceVersion)
+		objectAfterApplyUnstructured := &unstructured.Unstructured{}
+		err = env.GetScheme().Convert(objectAfterApply, objectAfterApplyUnstructured, nil)
+		g.Expect(err).ToNot(HaveOccurred())
```

Comment:
```suggestion
```
This is not necessary (anymore), right?
We can just use the resourceVersion from objectAfterApply